src/beacon.cpp

```diff
@@ -138,41 +138,82 @@ std::string RetrieveBeaconValueWithMaxAge(const std::string& cpid, int64_t iMaxS
 
 bool VerifyBeaconContractTx(const std::string& txhashBoinc)
 {
+    // Mandatory block needed.
+    // Current bad contracts in chain would cause a fork on sync
+    // Previous beacons in the past have 3 elements not 4 thus causing empty contract elements as coded in GetBeaconElements
+    if (
+               (fTestNet && nBestHeight <= 378000)
+            || (!fTestNet && nBestHeight <= 1070000)
+       )
+        return true; // allow sync from 0 till these blocks when this check comes into effect
```

Comment:
This is not necessary. Incoming loose transactions should be filtered from day 1. This checks should be only skipped for transactions in blocks before the mandatory.

Reply:
if u sync from zero its checking the beacon contracts and old beacons were 3 not 4 items. caused a halt right away at block 650ish. thats why i added the blocks for mandatory and i did my calc roughly on 14 days. since mandatories needs give exchanges 14 days. but ravon can change it